File: leetcode2042.java
class Solution {
    public boolean areNumbersAscending(String s) {
        String[] s1 = s.split(" ");
        ArrayList<Integer> nums = new ArrayList<>();
        Arrays.stream(s1).forEach((s2) -> {
            if (s2.charAt(0) >= '0' && s2.charAt(0) <= '9') {
                int i = Integer.parseInt(s2);
                nums.add(i);
            }
        });
        for (int i = 1;i < nums.size();i ++) {
            if (nums.get(i) <= nums.get(i - 1)) {
                return false;
            }
        }
        return true;
    }
}

File: leetcode2042.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool areNumbersAscending(string s) {
        vector<string> val;
        for (int i = 0; i < s.size(); i++) {
            string x;
            while (s[i] != ' ')
                x.push_back(s[i ++]);
            val.push_back(x);
        }
        vector<int> nums;
        std::for_each(val.begin(), val.end(), [&](basic_string<char> item) {
            if (item[0] >= '0' && item[0] <= '9')
                nums.push_back(atoi(item.c_str()));
        });
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] < nums[i - 1])
                return false;
        }
        return true;
    }
};

int main() {
    string s = "1 box has 3 blue 4 red 6 green and 12 yellow marbles";
    cout << (new Solution)->areNumbersAscending(s);
    return 0;
}